src/sourceInfo.ts

```diff
@@ -0,0 +1,88 @@
+import { google } from '../build/pbjs';
+import FileDescriptorProto = google.protobuf.FileDescriptorProto;
+
+export type FieldID = number;
+
+export const Fields = {
+    file: {
+        syntax: 12,
+        message_type: 4,
+        enum_type: 5,
+        service: 6,
+        extension: 7,
+    },
+    message: {
+        field: 2,
+        nested_type: 3,
+        enum_type: 4,
+    },
+    enum: {
+        value: 2,
+    },
+    service: {
+        method: 2,
+    },
+}
+
+export interface SourceDescription { 
+    readonly span: number[];
+    readonly leadingComments: string;
+    readonly trailingComments: string;
+    readonly leadingDetachedComments: string[];
+};
+
+class EmptyDescription implements SourceDescription {
+    span = [];
+    leadingComments = '';
+    trailingComments = '';
+    leadingDetachedComments = [];
+};
+
+export type SourceInfoMap = {
+    [key: string]: SourceDescription
```

Comment:
Maybe add an in-source comment about what some example keys are?

Reply:
@stephenh Done.